internal/alertmanager/upstream.go

```diff
@@ -82,15 +87,31 @@ func GetAlertmanagerByName(name string) *Alertmanager {
 // WithProxy option can be passed to NewAlertmanager in order to enable request
 // proxying for unsee clients
 func WithProxy(proxied bool) Option {
-	return func(am *Alertmanager) {
+	return func(am *Alertmanager) error {
 		am.ProxyRequests = proxied
+		return nil
 	}
 }
 
 // WithRequestTimeout option can be passed to NewAlertmanager in order to set
 // a custom timeout for Alertmanager upstream requests
 func WithRequestTimeout(timeout time.Duration) Option {
-	return func(am *Alertmanager) {
+	return func(am *Alertmanager) error {
 		am.RequestTimeout = timeout
+		return nil
+	}
+}
+
+// WithTLSConfig option can be passed to NewAlertmanager in order to set
+// custom options for the TLS handling when sending requests to Alertmanager
+// upstream
+func WithTLSConfig(tlsOptions transport.TLSConnectionOptions) Option {
```

Comment:
Either rename this function to `WithTLSOptions` or similar or do this work to turn the YAML options into a standard `tls.Config` and pass that here instead.

Reply:
I think I'm gonna go with dedicated option for setting CA certs and client cert/key